src/Core/Configurations/RuntimeConfigProvider.cs

```diff
@@ -40,14 +40,13 @@ public class RuntimeConfigProvider
     /// </summary>
     public string? ManagedIdentityAccessToken { get; private set; }
 
-    private readonly RuntimeConfigLoader _runtimeConfigLoader;
-    private RuntimeConfig? _runtimeConfig;
+    public RuntimeConfigLoader ConfigLoader { get; private set; }
```

Comment:
What's the reason that this is public? There shouldn't be a need for anything that uses the Provider to directly access the Loader.

Reply:
There was code in startup that depends on showing error logs based on where the runtimeconfig filepath is. I did not want to alter that behaviour. Since Runtimeconfigprovider now has a generic runtimeconfigloader and not a filesystemloader, there was no way to get the filepath without getting the loader.

Reply:
Yeah I did end up seeing it (and commented on it later in the PR), I was just lazy and didn't delete this comment 😅